aiokafka/consumer.py

```diff
@@ -1,375 +1,612 @@
 import asyncio
-from collections import deque
-
 import logging
-import numbers
-from itertools import zip_longest, repeat
 
-from kafka.common import (
-    OffsetRequest, OffsetCommitRequest, OffsetFetchRequest,
-    UnknownTopicOrPartitionError, FetchRequest,
-    ConsumerFetchSizeTooSmall, ConsumerNoMoreData,
-    check_error
-)
+from kafka.common import OffsetAndMetadata, TopicPartition
+from kafka.coordinator.assignors.roundrobin import RoundRobinPartitionAssignor
+from kafka.consumer.subscription_state import SubscriptionState
+
+from aiokafka.client import AIOKafkaClient
+from aiokafka.group_coordinator import GroupCoordinator
+from aiokafka.fetcher import Fetcher
+from aiokafka import __version__
+
+log = logging.getLogger(__name__)
+
+
+class AIOKafkaConsumer(object):
+    """Consume records from a Kafka cluster.
+
+    The consumer will transparently handle the failure of servers in the Kafka
+    cluster, and adapt as topic-partitions are created or migrate between
+    brokers. It also interacts with the assigned kafka Group Coordinator node
+    to allow multiple consumers to load balance consumption of topics (requires
+    kafka >= 0.9.0.0).
+
+    Arguments:
+        *topics (str): optional list of topics to subscribe to. If not set,
+            call subscribe() or assign() before consuming records.
+        bootstrap_servers: 'host[:port]' string (or list of 'host[:port]'
+            strings) that the consumer should contact to bootstrap initial
+            cluster metadata. This does not have to be the full node list.
+            It just needs to have at least one broker that will respond to a
+            Metadata API Request. Default port is 9092. If no servers are
+            specified, will default to localhost:9092.
+        client_id (str): a name for this client. This string is passed in
+            each request to servers and can be used to identify specific
+            server-side log entries that correspond to this client. Also
+            submitted to GroupCoordinator for logging with respect to
+            consumer group administration. Default: 'aiokafka-{version}'
+        group_id (str or None): name of the consumer group to join for dynamic
```

Comment:
I really wander why would we set some string as default. I think it is more rational to set `None` as default here...